### PR TITLE
line chart: fix the nice domain for small constant

### PR DESCRIPTION
**Pull request by stephanwlee**
Also, this change improves testing around `niceDomain` and we assert on
non-positive value clipping.

This change partially address #4362 for the new gpu line chart. We will later
integrate the new scale to the legacy vz_line_chart.

### CODE REVIEW

File: tensorboard/webapp/widgets/line_chart_v2/lib/scale.ts [hunk 1]
```diff
@@ -74,7 +74,9 @@ class LinearScale implements Scale {
     const padding =
       max === min
         ? // In case both `min` and `max` are 0, we want some padding.
-          Math.max(min * PADDING_RATIO, MIN_SIGNIFICANT_PADDING)
+          min === 0
+          ? PADDING_FOR_ZERO
+          : Math.abs(min) * PADDING_RATIO
```
Comment (nfelt) [inline tensorboard/webapp/widgets/line_chart_v2/lib/scale.ts:79]: FWIW, I'm ok with this padding approach, but historically we had added more generous padding for constant plots (basically using `[0, 2c]` as the range) to ensure that we were always showing the x axis, while still adjusting the overall scale to match the magnitude of the constant.  Did you have a preference for the tighter zoom?  Could see an argument either way, just curious.

**Review by nfelt — APPROVED** [review]
Thanks for the fix!

File: tensorboard/webapp/widgets/line_chart_v2/lib/scale.ts (hunk 1 above)
Comment (stephanwlee) [reply tensorboard/webapp/widgets/line_chart_v2/lib/scale.ts]: I see some reference to that time, but I am not sure whether that's what we have today (comment does not agree with the code, at least not in an obvious way).

https://github.com/tensorflow/tensorboard/blob/c7a4c73ca389318b200de157dda253f07279eeb8/tensorboard/components/vz_line_chart2/linear-scale.ts#L45-L50 

Regardless of what is happening in vz_line_chart, decided to include `0` in all cases.

File: tensorboard/webapp/widgets/line_chart_v2/lib/scale.ts [hunk 2]
```diff
@@ -161,10 +164,7 @@ class Log10Scale implements Scale {
       spreadInLog > 0
         ? spreadInLog * PADDING_RATIO
         : // In case `minLogValue` is 0 (i.e., log_10(1) = 0), we want some padding.
-          Math.max(
-            Math.abs(minLogValue * PADDING_RATIO),
-            MIN_SIGNIFICANT_PADDING
-          );
+          Math.max(Math.abs(minLogValue * PADDING_RATIO), PADDING_FOR_ZERO);
```
[the remark this answers is not in the document]
Comment (stephanwlee) [reply tensorboard/webapp/widgets/line_chart_v2/lib/scale.ts]: Put some constant padding in the log scale (halving and doubling in log scale is just `log_10(x) ± log_10(2)`. I am not sure if it is visually pleasing just yet.

File: tensorboard/webapp/widgets/line_chart_v2/lib/scale.ts [hunk 3]
```diff
@@ -148,28 +149,13 @@ class Log10Scale implements Scale {
...
-          Math.max(
-            Math.abs(minLogValue * PADDING_RATIO),
-            MIN_SIGNIFICANT_PADDING
-          );
-
-    return [
-      this.untransform(Math.max(numericMinLogValue, minLogValue - padInLog)),
-      this.untransform(maxLogValue + padInLog),
-    ];
+    return [Math.max(Number.MIN_VALUE, adjustedMin * 0.5), adjustedMax * 1.5];
```
Comment (nfelt) [inline tensorboard/webapp/widgets/line_chart_v2/lib/scale.ts:158]: You mentioned "Put some constant padding in the log scale (halving and doubling in log scale...)" but this is halving and then adding 50% - should it be `adjustedMax * 2.0`?  Or do we want the upper padding to be smaller for some reason?

This also seems like it doesn't match the tests which might be causing the failures (although weirdly, only for one of the two travis jobs).

**Review by nfelt — APPROVED** [review]
LGTM.  Yeah I agree the code in `vz-line-chart` is hard to parse, but the new code here looks great.

File: tensorboard/webapp/widgets/line_chart_v2/lib/scale.ts (hunk 3 above)
Comment (stephanwlee) [reply tensorboard/webapp/widgets/line_chart_v2/lib/scale.ts]: Yup, that was a mistake on my part. Fixed the test and code.
